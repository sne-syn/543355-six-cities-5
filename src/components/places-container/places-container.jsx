import React from 'react';
import PropTypes from 'prop-types';
import Sort from '../sort/sort';
import MapSection from '../map-section/map-section';
import ListCities from '../list-cities/list-cities';
import {CountCards} from '../../utils/const';

const PlacesContainer = ({filteredOffers, currentCity}) => {
  return (
    <div className="cities__places-container container">
      <section className="cities__places places">
        <h2 className="visually-hidden">Places</h2>
        <b className="places__found">{filteredOffers.length} {filteredOffers.length === 1 ? `place` : `places`} to stay in {currentCity}</b>
        <Sort />
        <ListCities listClass={`cities__places-list tabs__content places__list`} filteredOffers={filteredOffers} countCards={CountCards.MAIN_LIST} />
      </section>
      <div className="cities__right-section">
        <section className="cities__map map">
          <MapSection currentCity={currentCity} filteredOffers={filteredOffers}/>
        </section>
      </div>
    </div>
  );
};

PlacesContainer.propTypes = {
  filteredOffers: PropTypes.array.isRequired,
  currentCity: PropTypes.string.isRequired,
};

export default PlacesContainer;
