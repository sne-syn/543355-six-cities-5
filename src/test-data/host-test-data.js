export const hostPro = {
  avatar: `https://robohash.org/61?set=set2&size=74x74`,
  id: 5,
  isPro: true,
  name: `Janette`,
};
export const host = {
  avatar: `https://robohash.org/61?set=set2&size=74x74`,
  id: 5,
  isPro: false,
  name: `Janette`,
};
