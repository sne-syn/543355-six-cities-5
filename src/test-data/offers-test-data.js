export const offers = [{
  bedrooms: 1,
  city: {
    location: {
      latitude: 52.370216,
      longitude: 4.895168,
      zoom: 10,
    },
    name: `Paris`,
  },
  description: `Gran suite con entrada independiente y baño interior privado en el centro de Segovia. Privacidad y comodidad a tan sólo tres minutos del acueducto.Amplia y luminosa, con 2 camas de 105 (+ 1 opcional de 90), baño completo, calefacción, mininevera, microondas, vajilla, cubertería básica, Internet y un balcón grande y soleado.I speak English, Italian and Spanish`,
  goods: [`cabel TV`, `heating`, `kitchen`, `towels`, `washing machine`, `washer`, `dishwasher`, `conditioner`, `elevator`, `coffee machine`],
  host: {
    avatar: `https://robohash.org/28?set=set2&size=74x74`,
    id: 4,
    isPro: false,
    name: `Salom`,
  },
  id: 1,
  previewImage: `https://bit.ly/34LWdhj`,
  images: [`https://bit.ly/34LWdhj`, `https://bit.ly/36Ys6Wn`, `https://bit.ly/33Qpecj`, `https://bit.ly/30YVQif`, `https://bit.ly/36TgIv7`, `https://bit.ly/3lEt2DG`],
  isFavorite: true,
  isPremium: false,
  location: {latitude: 48.865610000000004,
    longitude: 2.350499,
    zoom: 5},
  maxGuests: 8,
  price: 207,
  rating: 1.4018800815887484,
  title: `La Latina- Small Charming Studio`,
  type: `room`,
}, {
  bedrooms: 3,
  city: {
    location: {
      latitude: 52.370216,
      longitude: 4.895168,
      zoom: 10,
    },
    name: `Brussels`,
  },
  description: `In full nature and very close to public transport. Green area surrounded by all kinds of sports facilities including a golf course, paddle, artificial grass soccer, pelota court. In an urbanization of villas with landscaped plots of about 500 m2`,
  goods: [`elevator`, `dishwasher`, `coffee machine`, `washer`, `washing machine`],
  host: {
    avatar: `https://robohash.org/82?set=set2&size=74x74`,
    id: 6,
    isPro: false,
    name: `Whitney`,
  },
  id: 2,
  previewImage: `https://bit.ly/34LWdhj`,
  images: [`https://bit.ly/30YVQif`, `https://bit.ly/3lEt2DG`, `https://bit.ly/36TgIv7`, `https://bit.ly/33Qpecj`, `https://bit.ly/36Ys6Wn`, `https://bit.ly/34LWdhj`],
  isFavorite: false,
  isPremium: false,
  location: {latitude: 48.865610000000004,
    longitude: 2.350499,
    zoom: 5},
  maxGuests: 5,
  price: 194,
  rating: 1.9069547983791004,
  title: `Estudio/suite completo en el centro de Segovia`,
  type: `house`,
}];
