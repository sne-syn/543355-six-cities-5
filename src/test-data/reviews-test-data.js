export const review = {
  author: `Beaulah`,
  avatar: `https://robohash.org/54?set=set2&size=54x54`,
  date: `2004-09-06T16:12:32.554Z`,
  id: 0,
  rating: 1,
  comment: `Antoni is a great host and us a wonderful home. His suggestions for places to go were killer!10/10 would stay again.`,
};

export const reviews = [{
  author: `Ione`,
  avatar: `https://robohash.org/98?set=set2&size=54x54`,
  date: `2007-10-13T16:12:32.554Z`,
  id: 1,
  rating: 4,
  comment: `There’s a stunning beach with great snorkelling just a 6min drive, a few other beautiful beaches close by too.`,
}, {
  author: `Bruno`,
  avatar: `https://robohash.org/53?set=set2&size=54x54`,
  date: `2005-08-25T16:12:32.554Z`,
  id: 2,
  rating: 3,
  comment: `Maria’s place is at a perfect location, very easy to get around with public transportation and walking distance to awesome bars and restaurants. Her place is great, very clean and has everything you need.`,
}];
