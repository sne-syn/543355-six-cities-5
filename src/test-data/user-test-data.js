export const user = {
  userAvatar: `https://robohash.org/82?set=set2&size=74x74`,
  userEmail: `trust_no_1@gmail.com`,
  userId: 1,
  userIsPro: false,
  userName: `fox_mulder`
};
