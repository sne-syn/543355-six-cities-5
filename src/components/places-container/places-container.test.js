import React from 'react';
import renderer from 'react-test-renderer';
import {PlacesContainer} from './places-container';
import {offers} from '../../test-data/offers-test-data';

jest.mock(`../list-hover-on-map/list-hover-on-map`, () => `ListHoverOnMap`);
jest.mock(`../map-section/map-section`, () => `MapSection`);
jest.mock(`../sort/sort`, () => `Sort`);
jest.mock(`../../hocs/with-toggler/with-toggler`, () => (Component) => (props) => <Component {...props} toggleComponent={()=>{}} on={false} />);

describe(`<PlacesContainer /> test`, () => {
  it(`renders PlacesContainer with multiple offers`, () => {
    const tree = renderer
    .create(
        <PlacesContainer
          activeElement={`Paris`}
          unsortedOffers={offers}
          highlightedOfferId={`5`}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
