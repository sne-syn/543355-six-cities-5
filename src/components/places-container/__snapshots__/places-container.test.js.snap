// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlacesContainer /> snapshots renders PlacesContainer with multiple offers 1`] = `
<div
  className="cities__places-container container"
>
  <section
    className="cities__places places"
  >
    <h2
      className="visually-hidden"
    >
      Places
    </h2>
    <b
      className="places__found"
    >
      0
       
      places
       to stay in 
      Paris
    </b>
    <Sort
      defaultOnValue={false}
    />
    <ListHoverOnMap
      type="cities"
    />
  </section>
  <div
    className="cities__right-section"
  >
    <section
      className="cities__map map"
    >
      <MapSection
        activeCity="Paris"
        activeOffer={5}
        offersToShowOnMap={Array []}
      />
    </section>
  </div>
</div>
`;
